lib/core/utils/query-selector-all-filter.js

```diff
@@ -34,23 +34,24 @@ function matchExpressions(domTree, expressions, filter) {
   let currentLevel = createLocalVariables(
     vNodes,
     expressions,
-    [],
+    null,
     domTree[0].shadowId,
     recycledLocalVariables.pop()
   );
   const result = [];
 
   while (currentLevel.vNodesIndex < currentLevel.vNodes.length) {
     const vNode = currentLevel.vNodes[currentLevel.vNodesIndex++];
-    const childOnly = []; // we will add hierarchical '>' selectors here
-    const childAny = [];
+    let childOnly = null; // we will add hierarchical '>' selectors here
+    let childAny = null;
     const combinedLength =
-      currentLevel.anyLevel.length + currentLevel.thisLevel.length;
+      (currentLevel.anyLevel?.length || 0) +
+      (currentLevel.thisLevel?.length || 0);
     let added = false;
     // see if node matches
     for (let i = 0; i < combinedLength; i++) {
       const exp =
-        i < currentLevel.anyLevel.length
+        i < (currentLevel.anyLevel?.length || 0)
           ? currentLevel.anyLevel[i]
           : currentLevel.thisLevel[i - currentLevel.anyLevel.length];
```

Comment:
This will error if `anyLevel` is null but `thisLevel` isn't

Reply:
Fixed.